eclair-core/src/main/scala/fr/acinq/eclair/api/JsonSerializers.scala

```diff
@@ -132,3 +133,17 @@ class DirectionSerializer extends CustomSerializer[Direction](format => ({ null
   case d: Direction => JString(d.toString)
 }))
 
+class PaymentRequestSerializer extends CustomSerializer[PaymentRequest](format => ({ null },{
+  case p: PaymentRequest => JObject(JField("prefix",JString(p.prefix) ) ::
+      JField("amount",JLong(p.amount.getOrElse(MilliSatoshi(-1L)).toLong) ) ::
+      JField("timestamp",JLong(p.timestamp) ) ::
+      JField("nodeId",JString(p.nodeId.toString()) ) ::
+      JField("description",JString(
+          p.description match{ case Left(l) => l.toString()
+                               case Right(r) => r.toString() }
+          ) ) ::
+      JField("paymentHash",JString(p.paymentHash.toString()) ) ::
+      JField("expiry",JLong(p.expiry.getOrElse(-1L)) ) ::
```

Comment:
Perhaps it's better to use a JNull here? Seems more correct to return a json `null` Instead of a dummy -1 in the case where the value is missing.

Reply:
Fixed